test-vectors/secret-tree.json

```diff
@@ -0,0 +1,2284 @@
+[
+  {
+    "cipher_suite": 1,
+    "n_leaves": 1,
+    "encryption_secret": "98e17a91f59669d7a64bcfa706e3fe5c0bf0973c807c8aead326cc4dc368da1d",
+    "sender_data": {
+      "sender_data_secret": "530eb44c6d216b3ecd6c9043fe4bae2b0ca46411bc27914e5519eb8eb352d43a",
+      "ciphertext": "7fb6d93ee3e4ed5fe44b129fd73ae4b528feef3ed71c918a32be983f67d577ec186ec726b0d42fab07aa1485a00c0a2c402a023a6037074eea3f6dbf84c62245316261cd15f4d3cde7f6264e4d",
+      "key": "dc0a4a17f3fcebae06c56b25448aef95",
+      "nonce": "ebda7c8d58ce2b2c95024570"
+    },
+    "leaves": [
+      {
+        "generation": 15,
+        "application_key": "d4f7ae974e8814bfb9f5afd60d691f74",
+        "application_nonce": "cd8d974348c3d739ae028726",
+        "handshake_key": "245bb77d170c9152f5093e436d76807c",
+        "handshake_nonce": "7259803ca33e47c942976c58"
+      }
+    ]
+  },
+  {
+    "cipher_suite": 1,
+    "n_leaves": 8,
+    "encryption_secret": "01d0e99e8944f05e1ffba9c55a924aa3263fc2f736804ca6e18abebd9cfcf2a0",
+    "sender_data": {
+      "sender_data_secret": "7e38bcb38d60f0429dd21e3d2c0f946393aa48d94ca50dccd26b26024aaeaf17",
+      "ciphertext": "404a67a40161a5002a00a2f756c2af62b4b162d569c34d7649b36cd1f4e3b23416644ed0cbcbdfcfa6bcc457df9ee871e7a1c0ef2944bab5bece1a2ed3f94287b9839be19af38af9c22ea73635",
+      "key": "451be7498f6c8475d2d14f5c2622b41f",
+      "nonce": "f604e6e95af3536e722fc18d"
+    },
+    "leaves": [
+      {
+        "generation": 15,
+        "application_key": "2295064552a7c387bcb2902876f1a6cb",
+        "application_nonce": "4dd0a464188d522c50211d59",
+        "handshake_key": "8e82d0c1ad6f44ac65465658be9365f4",
+        "handshake_nonce": "78f0edef08fc96cb6a030e42"
+      },
```

Comment:
You should have an array of generations for each leaf not just a single generation.